### PR TITLE
Refactoring to es6 class syntax

### PR DESCRIPTION
**Pull request by gimdongwoo**
Thank you for your react-chatview.

I did refactoring react-chatview to use it in my project.

I have applied the eslint rule of air-bnb. Can I send anoter PR the setting of this part?

### CODE REVIEW

**Comment by dustingetz** [comment]
Hi gimgdongwoo, Thank you for your contribution. It's really hard to merge large changes like this. It's important to keep the change surface area as small as possible, so it is easy for people who have forked to contribute their changes back. Sorry :(

**Comment by dustingetz** [comment]
I will accept a PR that fixes the warnings with the minimum diff. When react fiber comes out, if substantial changes are necessary, that would be a good time to migrate to ES6 classes.

**Comment by gimdongwoo** [comment]
Recently, I used this version of PR for production, and I had no problems.
We'll update the PR by adding some changes to the part about PropTypes.

**Comment by dustingetz** [comment]
You're of course welcome to fork - thanks for using react-chatview! - but be careful, there are outstanding PRs against master that fix issues that will be landed eventually, so you'll need to backport them if you fork.

**Comment by gimdongwoo** [comment]
This PR is changed only the grammar without changing the logic itself.
Less changes than this PR would not fit React's needs.

**Comment by franciscop** [comment]
@gimdongwoo why does it need `stage-0` preset? From the docs and the code diff it doesn't seem to use it.

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "presets": ["react", "es2015"]
+  "presets": ["react", "es2015", "stage-0"],
```
Comment (franciscop) [inline .babelrc]: Does it need this preset? It doesn't seem used at all in here

Comment (gimdongwoo) [inline .babelrc]: Right! Good catch! Thank you.

Comment (gimdongwoo) [inline .babelrc]: It need to for this.

static propTypes = { ... }

Comment (gimdongwoo) [inline .babelrc]: It need to for this.

`static propTypes = { ... }`

I did change 'stage-0' preset to 'transform-class-properties' plugins.

Good catch! Thank you!

**Comment by gimdongwoo** [comment]
I am currently maintaining the ES6 version of the code separately, but I will clean up and merge it.

https://github.com/gimdongwoo/react-chatview-es6

**Pull request closed without merging**